odoo/addons/base/ir/ir_actions.py

```diff
@@ -385,8 +385,7 @@ class IrActionsActWindowView(models.Model):
     @api.model_cr_context
     def _auto_init(self):
         res = super(IrActionsActWindowView, self)._auto_init()
-        self._cr.execute('SELECT indexname FROM pg_indexes WHERE indexname = \'act_window_view_unique_mode_per_action\'')
-        if not self._cr.fetchone():
+        if not tools.index_exists(self._cr, 'act_window_view_unique_mode_per_action'):
             self._cr.execute('CREATE UNIQUE INDEX act_window_view_unique_mode_per_action ON ir_act_window_view (act_window_id, view_mode)')
```

Comment:
Good suggestion, I am going to give it a try.
I simply want to avoid making the API too complex.